### DISCUSSION TITLE
quantizeImage 10x slower on newer versions of ImageMagick

### DISCUSSION THREAD

**Discussion by c0dehunter** [ImageMagick/ImageMagick#2177] [Help]
I am using [$imagick->quantizeImage](https://phpimagick.com/Imagick/quantizeImage) in order to reduce image size as it only contains some lines in same color (like a mini chart). But with `ImageMagick 7.0.10-10 Q16 x86_64 2020-05-12 (alt-ImageMagick-7.0.10.10-1.el6.x86_64)` this function is about 10x slower than it was in previous versions. 

Example with 100 images:
- without `$imagick->quantizeImage(4, \Imagick::COLORSPACE_RGB, 0, false, false);`
![image](https://user-images.githubusercontent.com/1484658/85190625-a8120280-b2ec-11ea-9d6b-1657c36aabb0.png)

- with `$imagick->quantizeImage(4, \Imagick::COLORSPACE_RGB, 0, false, false);`
![image](https://user-images.githubusercontent.com/1484658/85190628-ae07e380-b2ec-11ea-9525-366ebaedb0e4.png)

How could I get same results while still having similar execution time as before?

**--- EDIT: ---**

I've noticed that turning on `dither` flag fixes this (the execution time is back to expected), e.g., use 

`$imagick->quantizeImage(4, \Imagick::COLORSPACE_RGB, 0, true, false);`
instead of 
`$imagick->quantizeImage(4, \Imagick::COLORSPACE_RGB, 0, false, false);`

**Comment by urban-warrior** [comment]
Post a one or two of your images or a link to your images.  We need to download them to reproduce the performance problem.  In the mean-time, use dither to speed up results.

**Comment by c0dehunter** [comment]
Hi urban-warrior, here are two sample images. We simply draw a line on a transparent background.

![test2](https://user-images.githubusercontent.com/1484658/85238404-3d81d380-b460-11ea-9695-8c7c49db2fe5.png)
![test1](https://user-images.githubusercontent.com/1484658/85238406-3eb30080-b460-11ea-97a4-02ce244d773b.png)

**Comment by urban-warrior** [comment]
Got it.  Thanks.  It will take a week or so before we can investigate and offer a solution.  Stand by.

**Comment by c0dehunter** [comment]
No worries, thank you for your help!

**Comment by urban-warrior** [comment]
Can you try with the latest release of ImageMagick?  We don't have IMagick installed but we converted 100 clones of your image into a 4 color, non-dithered, GIF animation from the command line.  It ran in .14 seconds.

```
convert sparkLine.png sparkLine.png ... +dither -colors 4 sparkLine.gif
```

With dithering enabled, it completed in 1.2 seconds.

**Reply by c0dehunter** [reply]
Could this be a problem with PHP extension IMagick? With Dither enabled, the processing times are ok. But if we DISABLE dither (which we want), it's 10x slower. Is there any similar function to quantizeImage which we could use for now? We just want to keep the yellow line and make the PNG size small.

I guess it would be more appropriate to ask the maintainers of PHP extension:

https://github.com/Imagick/imagick/issues/330
